src/main/java/com/iexec/common/utils/FeignBuilder.java

```diff
@@ -0,0 +1,58 @@
+package com.iexec.common.utils;
+
+import feign.Feign;
+import feign.Logger;
+import feign.RequestTemplate;
+import feign.Response;
+import feign.codec.Encoder;
+import feign.codec.StringDecoder;
+import feign.jackson.JacksonDecoder;
+import feign.jackson.JacksonEncoder;
+import feign.slf4j.Slf4jLogger;
+
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+/**
+ * Returns a Feign builder with shared configuration to create various Feign client instances.
+ * <p>
+ * The shared configuration is:
+ * <ul>
+ * <li>JacksonEncoder with overridden encode method to deal with strings
+ * <li>JacksonDecoder with overridden decode method to deal with strings
+ * <li>SLF4J logger enabled with FULL details. Logs are displayed by configuring the log level on a target interface logger.
+ * <li>application/json Content-Type header
+ * </ul>
+ */
+public class FeignBuilder {
+
+    private FeignBuilder() {};
+
+    public static Feign.Builder createBuilder(Logger.Level logLevel) {
+        return Feign.builder()
+                .encoder(new JacksonEncoder() {
```

Comment:
Why do we need these encoder and decoder? Could you add a comment to explain the reason?

Reply:
Documentation improved with [09297f0](https://github.com/iExecBlockchainComputing/iexec-common/pull/307/commits/09297f087e853ac857a439c0c932f0cee99a483c)

Reply:
Thanks!